radis/lbl/base.py

```diff
@@ -3239,64 +3240,108 @@ def plot_populations(self, what="vib", isotope=None, nfig=None):
+            raise ValueError("Please use wmin/wmax when passing wunit")
+        if isinstance(wavelength_min, u.Quantity):
+            assert isinstance(wavelength_min, u.Quantity) and isinstance(
+                wavelength_max, u.Quantity
+            )
+            assert wavelength_min.unit.is_equivalent(u.m)
+            assert wavelength_max.unit.is_equivalent(u.m)
 
-        # In wavelength mode, the propagating medium matters. Convert to
-        # calculation medium (vacuum) if needed:
+    if wavenum_min is not None or wavenum_max is not None:
+        assert wavenum_min is not None and wavenum_max is not None
+        assert wavenum_min < wavenum_max
+        if not isinstance(wunit, Default):
+            raise ValueError("Please use wmin/wmax when passing wunit")
+        if isinstance(wavenum_min, u.Quantity) or isinstance(wavenum_max, u.Quantity):
+            assert isinstance(wavenum_min, u.Quantity) and isinstance(
+                wavenum_max, u.Quantity
+            )
+            assert wavenum_min.unit.is_equivalent(1 / u.cm)
+            assert wavenum_max.unit.is_equivalent(1 / u.cm)
+
+    if isinstance(wmin, u.Quantity) or isinstance(wmax, u.Quantity):
+        assert wmin is not None and wmax is not None
+        assert isinstance(wmin, u.Quantity) and isinstance(wmax, u.Quantity)
+        assert wmin.unit.is_equivalent(u.m) or wmin.unit.is_equivalent(1 / u.m)
+        assert wmax.unit.is_equivalent(u.m) or wmax.unit.is_equivalent(1 / u.m)
+        assert wmin.unit.is_equivalent(wmax.unit)
+        if not isinstance(wunit, Default):
+            if not wmin.unit.is_equivalent(u.Unit(wunit)):
+                raise ValueError("Conflicting units passed for wmin/wmax and wunit")
+            # Should I keep this in?
+            # Deals with cases like: calc_spectrum(wmin=10*u.cm, wmax=1.u.m, wunit="cm")
+    #             else:
+    #                 if wmin.unit != wmax.unit and (wmin.unit != u.Unit(wunit.value) or wmax.unit != u.Unit(wunit.value)):
+    #                     raise Warning("Ambiguous units passed, ignoring wunit")
+
+    if wmin is not None and wmax is not None:
+        if isinstance(wmin, u.Quantity) or isinstance(wmax, u.Quantity):
+            if wmin.unit.is_equivalent(u.m):
+                wavelength_min = wmin
+                wavelength_max = wmax
+            else:
+                wavenum_min = wmin
+                wavenum_max = wmax
+        else:
+            if isinstance(wunit, Default):
+                wavenum_min = wmin * u.Unit(wunit.value)
+                wavenum_max = wmax * u.Unit(wunit.value)
+            else:
+                if u.Unit(wunit).is_equivalent(u.m):
+                    wavelength_min = wmin * u.Unit(wunit)
+                    wavelength_max = wmax * u.Unit(wunit)
+                else:
+                    wavenum_min = wmin * u.Unit(wunit)
+                    wavenum_max = wmax * u.Unit(wunit)
+
+    if wavenum_min is not None or wavenum_max is not None:
```

Comment:
Normally we already checked that if one is given the other one too. This can simply be:
```suggestion
    if wavenum_min:
```